phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -1628,6 +1689,7 @@ private List<Mutation> generateOnDupMutations(BatchMutateContext context, Put at
               }
               // update the cells to the latest values calculated above
               flattenedCells = mergeCells(flattenedCells, updatedCells);
+              flattenedCells.sort(CellComparator.getInstance());
```

Comment:
Can you add a comment here why we need to sort ? I couldn't figure out.

Reply:
This is really interesting part :)
As part of `addEmptyKVCells()`, we need to get empty cell: `Cell emptyKVCell = tuple.getValue(familyBytes, emptyCQ);`

If we don't sort cells here, tuple#getValue might not be able to retrieve the right empty column cell because getValue() relies on binary search to get the right Cell.

Let's add a short comment here that "we need to retrieve empty cell later on"?

Reply:
Done